src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.GenerateNamedType.cs

```diff
@@ -224,12 +224,10 @@ private void AddProperties(ArrayBuilder<ISymbol> members)
                     parameterType = parameterType.RemoveUnavailableTypeParameters(
                         _semanticDocument.SemanticModel.Compilation, availableTypeParameters);
 
-                    if (!TryFindMatchingField(parameterName, parameterType, parameterToExistingFieldMap, caseSensitive: true))
+                    if (!TryFindMatchingField(parameterName, parameterType, parameterToExistingFieldMap, caseSensitive: true) &&
```

Comment:
It feels like finding the matching field based on the case sensitivity should be baked into this method instead of being passed as a parameter

Reply:
will try to make that change :)

Reply:
done!